apollo-runtime/src/main/java/com/apollographql/apollo/ApolloCall.kt

```diff
@@ -93,6 +94,15 @@ interface ApolloCall<D : Operation.Data> : Cancelable {
      */
     abstract fun onResponse(response: Response<D>)
 
+    /**
+     * Gets called when the GraphQL response has been cached successfully.
+     * It should be called only once from the cache.
+     * This is used internally by watchers to retrieve the list of dependentKeys they have to watch.
+     *
+     * @param records the [List] of [Record] that was merged in cache.
+     */
+    open fun onCached(records: List<Record>) {}
```

Comment:
For the record (pun not intented 😁 ), we discussed using a `Set` here instead of a `List` as `records` should be unique. But since we don't need random access in this list and `Lists` are more economical than `Sets`, it make sense to keep a `List`